apps/stripe/src/app/api/stripe/webhook/stripe-object-handlers/stripe-payment-intent-handler.test.ts

```diff
@@ -256,5 +260,48 @@ describe("StripePaymentIntentHandler", () => {
         },
       );
     });
+
+    describe("type: payment_intent.canceled", () => {
+      it.each([
+        {
+          resolvedTransactionFlow: createResolvedTransactionFlow("AUTHORIZATION"),
+          expectedType: "CANCEL_SUCCESS",
+        },
+        {
+          resolvedTransactionFlow: createResolvedTransactionFlow("CHARGE"),
+          expectedType: "CANCEL_SUCCESS",
+        },
+      ])(
+        "should resolve fields with type: $expectedType from Stripe event properly for $resolvedTransactionFlow flow",
+        ({ resolvedTransactionFlow, expectedType }) => {
+          const event = getMockedPaymentIntentPaymentCanceledEvent();
+
+          event.data.object.amount = 123_30;
+
+          const recordedTransaction = getMockedRecordedTransaction({
+            resolvedTransactionFlow,
+          });
+
+          const result = new StripePaymentIntentHandler().processPaymentIntentEvent({
+            event,
+            recordedTransaction,
+            stripePaymentIntentId: mockedStripePaymentIntentId,
+            stripeEnv: "LIVE",
+          });
+
+          const { type, amount, pspReference, time } = result
+            ._unsafeUnwrap()
+            .resolveEventReportVariables();
+
+          expect(type).toBe(expectedType);
+          // comes from mock
+          expect(amount.currency).toStrictEqual("USD");
+          // Converted to Saleor float
+          expect(amount.amount).toStrictEqual(123.3);
```

Comment:
lest do 

```
const amountToUse = 123_30
event.data.object.amount = amountToUse
expect(amount.amount).toStrictEqual(amountToUse);

```

It will be a litle more explicit that we use ACTUAL amount from event, best if our fixtures have different ones.

I want to avoid situation where tests pass because by accident amount is the same as the other amount, but we actually use wrong one